examples/qubit_characterizations_example.py

```diff
@@ -0,0 +1,47 @@
+from matplotlib import pyplot as plt
+import numpy as np
+import cirq
+
+
+def main():
+    # The device to run the experiment.
+    simulator = cirq.Simulator()
+
+    # The two qubits to be characterized in this example.
+    q_0 = cirq.GridQubit(0, 0)
+    q_1 = cirq.GridQubit(0, 1)
+
+    # Measure Rabi oscillation of q_0.
+    rabi_results = cirq.experiments.rabi_oscillations(simulator, q_0, 4 * np.pi)
+    rabi_results.plot()
+
+    # Clifford-based randomized benchmarking of single-qubit gates on q_0.
+    n_cfs_1q = range(10, 100, 10)
+    rb_result_1q = cirq.experiments.single_qubit_randomized_benchmarking(
+        simulator, q_0, n_cfs_1q)
+    rb_result_1q.plot()
+
+    # Clifford-based randomized benchmarking of two-qubit gates on q_0 and q_1.
+    n_cfs_2q = range(5, 50, 5)
+    rb_result_2q = cirq.experiments.two_qubit_randomized_benchmarking(
+        simulator, q_0, q_1, n_cfs_2q)
+    rb_result_2q.plot()
+
+    # State-tomography of q_0 after application of an X/2 rotation.
+    cir_1q = cirq.Circuit.from_ops(cirq.X(q_0) ** 0.5)
+    tomography_1q = cirq.experiments.single_qubit_state_tomography(simulator,
+                                                                   q_0, cir_1q)
+    tomography_1q.plot()
+
+    # State-tomography of a Bell state between q_0 and q_1.
+    cir_2q = cirq.Circuit.from_ops(cirq.H(q_0), cirq.CNOT(q_0, q_1))
+    tomography_2q = cirq.experiments.two_qubit_state_tomography(simulator,
+                                                                q_0, q_1,
+                                                                cir_2q)
+    tomography_2q.plot()
+
+    plt.show()
```

Comment:
Is this redundant now? Please add a screenshot of the output as a reply to this comment.

Reply:
Without the plt.show() command, all the figures immediately close after opening unless I'm in the interactive mode of PyCharm. I left it for now. The output of the two-qubit tomography function looks like this (both bar plots in the same figure):

![Screenshot from 2019-03-14 22-33-02](https://user-images.githubusercontent.com/41397361/54410625-81a4f680-46a9-11e9-8b70-56dc28b3f74c.png)

Reply:
Wait, you're saying that calling `plt.show()` twice (once inside `tomography_2q.plot()` and once after it) is on purpose? That doesn't make sense to me. Pycharm closes the first one but not the second one? How does it behave when executing outside of pycharm?

Reply:
Sorry I didn't make this clear. There are actually two kinds of `.show` method, one that goes after a `Figure` object and one that goes after `plt`. The `.show` within the various `plot` methods in the experimental result classes come after the `Figures` object, whereas the `.show` within the example file comes after `plt` itself.

Calling `.show` after a `Figure` object will show the figure but close it soon afterwards so the program can keep running, whereas calling `.show` after 'plt' will show and hold the figure, but halts the execution of the program until the user closes the figure manually. That's why there are two calls to '.show' in the code. I hope this makes sense? A related post on this is here [https://stackoverflow.com/questions/9280171/matplotlib-python-show-returns-immediately](url).

In principle, we can get rid of all the `show` within the plot methods inside the experimental result classes to avoid redundancy. Alternatively we can save the plot to some user-specified directory. Please let me know what you prefer.

Reply:
I'm not able to reproduce this flickering behavior.

It's acceptable to me if you use either `plt.show` or `fig.show`, as long as you only use one and it occurs inside the method.

Reply:
Done, I removed the plt.show() from the example.